vllm/executor/ray_gpu_executor.py

```diff
@@ -23,12 +22,28 @@
 
 logger = init_logger(__name__)
 
+# If the env var is set, it uses the Ray's compiled DAG API
+# which optimizes the control plane overhead.
+# Run vLLM with VLLM_USE_RAY_COMPILED_DAG=1 to enable it.
+# Currently, this requires USE_SPMD_WORKER=True.
```

Comment:
Should we remove it in this case?

Reply:
I think the plan is to support SPMD for non-DAG as well: https://github.com/vllm-project/vllm/pull/6032/files#diff-4fc454dabd93d2c8a43a962360afe9753d139415326bc3097c99815bf0311736R43
So two flags are desired. So far since only DAG is supported, we check the passed args to fail early.